### PR TITLE
Enable ibis.literal to detect and preserve when datetime has a timezone.

### LINKED ISSUE #1459: Timezone information is not passed from datetime objects through to ibis.literal

### PR DESCRIPTION
**Pull request by missing-semicolon**
Closes #1459

### CODE REVIEW

File: ibis/expr/tests/test_datatypes.py [hunk 1]
```diff
@@ -216,8 +230,8 @@ def test_timestamp_with_timezone_parser_invalid_timezone():
 
 
 @pytest.mark.parametrize('unit', [
-    'Y', 'Q', 'M', 'W',  'D',  # date units
```
Comment (missing-semicolon) [inline ibis/expr/tests/test_datatypes.py]: Looks like my linting caused some collateral damage. 💥

File: ibis/expr/datatypes.py [hunk 2]
```diff
@@ -1139,7 +1139,10 @@ def infer_date(value):
 
 @infer.register(datetime.datetime)
 def infer_timestamp(value):
-    return timestamp
+    if value.tzinfo:
+        return Timestamp(timezone=value.tzinfo)
```
Comment (cpcloud) [inline ibis/expr/datatypes.py]: We don't actually enforce this anywhere, but should we convert `value.tzinfo` to a `str` to be consistent with other uses of the "timestamp with timezone" type in the codebase?

**Comment by cpcloud** [comment]
Merging.

**Comment by cpcloud** [comment]
@missing-semicolon thank you sir!

**Pull request closed without merging**